lib/home_page.dart

```diff
@@ -0,0 +1,348 @@
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+
+import 'assets/constants.dart';
+import 'assets/enums.dart';
+import 'assets/response_object.dart';
+import 'assets/extensions.dart';
+import 'assets/helper_methods.dart';
+
+class HomePage extends StatefulWidget {
+  @override
+  State<HomePage> createState() => _HomePageState();
+}
+
+class _HomePageState extends State<HomePage> {
+  final _formKey = GlobalKey<FormState>();
+  final nameController = TextEditingController();
+  final _dobController = TextEditingController();
+  final _opinionController = TextEditingController();
+  bool _acceptTerms = false;
+  Gender? _selectedGender;
+  Country? _selectedCountry;
+
+  void _submitForm() {
+    if (_formKey.currentState!.validate()) {
+      _formKey.currentState?.save();
+      showDialog(
+        context: context,
+        builder: (BuildContext context) => _createAlertDialog(),
+      );
+    }
+  }
+
+  @override
+  void dispose() {
+    nameController.dispose();
+    _opinionController.dispose();
+    _dobController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    var headlineSmall = Theme.of(context).textTheme.headlineSmall;
```

Comment:
Why var?

Reply:
What is the difference between `final`, `var` and `const`?

Reply:
final and const are used to set a value to a variable only once. Var is used for the variable that changes its value. final is a runtime constant and const variable should be calculated at compile time.

Reply:
> Why var?

Changed to final.